pulsar-client-cpp/include/pulsar/MessageBuilder.h

```diff
@@ -76,6 +76,24 @@ class MessageBuilder {
     MessageBuilder& setEventTimestamp(uint64_t eventTimestamp);
 
     /**
+     * Specify a custom sequence id for the message being published.
+     * <p>
+     * The sequence id can be used for deduplication purposes and it needs to follow these rules:
+     * <ol>
+     * <li><code>sequenceId >= 0</code>
```

Comment:
Assertion for this - also why not use `uint64_t`

Reply:
Added check. Keeping `int64_t` for consistency with initialSequence id which can be negative.